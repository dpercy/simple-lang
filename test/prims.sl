+ ; printing
(+ 2 3) ; arity
(+ 2 3 4)
(+ 1)
(+)
(substring "foo" 0 2)
(substring "foo" 0)
(+ "foo" "bar") ; types
(string-append "foo" "bar")
(string-append 2 3)
