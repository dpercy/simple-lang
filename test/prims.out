+
5
(error "+: expected 2 arguments but got 3")
(error "+: expected 2 arguments but got 1")
(error "+: expected 2 arguments but got 0")
"fo"
(error "substring: expected 3 arguments but got 2")
(error "+: expected an integer")
"foobar"
(error "string-append: expected a string")
